### ISSUE TITLE
Instant Custom Shorten Not Working

### ISSUE THREAD

**Issue by johnyradio** [YOURLS/YOURLS#2251]
### Technical details

* YOURLS version: 1.7.2
* PHP version: 5.6.29 (FastCGI)

### Reproducible Bug Summary 

1. Log into my yourls admin screen 
2. Click "Tools"
3. Drag "Instant Custom Shorten" to bookmarks bar
4. Browse to any desired webpage
5. Click "Instant Custom Shorten" in bookmarks bar
6. See popup
7. Enter desired short name
8. Browse to my yourls admin screen 
9. The new short url i just created is not there

This is a bug because... _the bookmarklet does not create a short url_

**Comment by ozh** [comment]
This is a very typical case of "if that was broken, someone from the tens of thousands of people using YOURLS would have been reporting it ages ago".

This definitely works.

Maybe refine step 4 and see if the problem is the one reported in #1662.

Feel free to comment with additional details if applicable, see in particular https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps#use-your-browser-developer-tools

**Comment by johnyradio** [comment]
> This is a very typical case of "if that was broken, someone from the tens of thousands of people using YOURLS would have been reporting it ages ago".

Seems it's a typical case of "someone _did_ report it, and it's a known issue."
This definitely does _not_ work, as you explained in your [post](https://github.com/YOURLS/YOURLS/issues/1662#issuecomment-38379930)

**Comment by nemws1** [comment]
Bug still exists - at least for Firefox.  Works in Chrome, though.

**Comment by sokai** [comment]
AFAIK this _problem_ is related to a wrong/missing Content Security Policy (CSP) of the server (where YOURLS is running). ([Here](https://csp-evaluator.withgoogle.com/) you can test the CSP and [here](https://cheatsheetseries.owasp.org/cheatsheets/Content_Security_Policy_Cheat_Sheet.html) you will find more information.)

If there is someone who can and is willing to share working CSP (with Nginx), I would be delighted. #ty

**Comment by Istria1704** [comment]
I have the same or a similar problem. I just installed yourls using docker-compose. And I'm using Nginx Proxy Manager to forward "link.mydomain.com" to the ip and port of the server docker is running on.

It worked once or twice though. Then it stopped. 

- I added the Instant Shorten bookmark to my bookmarks bar.
- Browsed to a random site.
- Clicked the bookmark
- A wild popup appears
- entered "askjdhsakjdhsad" and hit enter
- another wild popup appears showing mydomain.cc/askjdhsakjdhsad
- hit enter again
- And there it was, added in yourls.

On later attempts, only the first popup shows up. But after clicking OK, the second popup doesn't show and nothing is added in yourls. I tried it in Edge and in Thorium (Chromium-based). The time it did work, was in Thorium.
